lib/parser/source/buffer.rb

```diff
@@ -95,9 +95,7 @@ def self.reencode_string(input)
         elsif detected_encoding == Encoding::BINARY
           input
         else
-          input.
-            force_encoding(detected_encoding).
-            encode(Encoding::UTF_8)
+          input.force_encoding(detected_encoding)
```

Comment:
The PR has been updated. But there are other questions to resolve first; see my comments in the GH issue.